libs/extensions/include/NanoEvent.hpp

```diff
@@ -18,7 +18,17 @@ class NanoEvent {
 
   std::shared_ptr<Event> GetEvent() { return event; }
 
-  std::shared_ptr<PhysicsObjects> GetAllMuons(float matchingDeltaR = 0.1);
+  std::shared_ptr<PhysicsObjects> GetDRMatchedMuons(float matchingDeltaR = 0.1);
+  std::shared_ptr<PhysicsObjects> GetOuterDRMatchedMuons(float matchingDeltaR = 0.1);
+  std::shared_ptr<PhysicsObjects> GetSegmentMatchedMuons(int minMatches = 0);
+  
+  std::shared_ptr<PhysicsObjects> GetAllMuonVertexCollections();
+  std::shared_ptr<PhysicsObjects> GetVerticesForMuons(std::shared_ptr<PhysicsObjects> muonCollection);
+  std::shared_ptr<PhysicsObjects> GetVertexForDimuon(std::shared_ptr<PhysicsObject> muon1, std::shared_ptr<PhysicsObject> muon2);
+  std::pair<std::shared_ptr<PhysicsObject>,std::shared_ptr<PhysicsObject>> GetLooseMuonsInVertex(std::shared_ptr<PhysicsObject> muonVertex);
+
+  bool IndexExist(std::shared_ptr<PhysicsObjects> objectCollection, float index, bool isDSAMuon=false);
+  float DeltaR(float eta1, float phi1, float eta2, float phi2);
```

Comment:
I found an error here from me merging from main. The function is here so that I can calculate Delta R using outerEta and outerPhi of the muons, but this part of the code disappeared during the merge. I will update it

Reply:
Yeah, my point was that the delta r between 2 muons (however it's defined) is not a property of the event, so it doesn't really belong here. It would be a property of a muon itself (how far are you from this other muon?), or there could be some muons' processor which you'd ask how far these two muons are apart.